test/compilable/pr9383.d

```diff
@@ -0,0 +1,26 @@
+// REQUIRED_ARGS: -preview=dip1000
+// https://github.com/dlang/dmd/pull/9383
+
+void test() @safe
+{
+    int[1] a = [1];
+    cartesianProduct(a[]);
+}
+
+auto cartesianProduct(RR...)(RR ranges)
+{
+    static struct Result
+    {
+        pragma(msg, RR);
+        RR current;
+
+        void popFront() scope @safe
+        {
+            foreach (ref r; current)
+            {
+            }
+        }
+    }
+
+    return Result(ranges);
+}
```

Comment:
I recommend giving the new tests a try: https://github.com/dlang/dmd/tree/master/test/unit.

Reply:
It's not a lexer/parser issue.

Reply:
You can run whatever kinds of tests you like there. It's not limited to lexer/parser.

Reply:
I'm guessing it would look something like this test [1] with the addition of running the semantic phases.

[1] https://github.com/dlang/dmd/blob/master/test/unit/self_test.d